pkg/types/vulnerability.go

```diff
@@ -12,20 +12,14 @@ type DetectedVulnerability struct {
 	PkgID            string         `json:",omitempty"` // It is used to construct dependency graph.
 	PkgName          string         `json:",omitempty"`
 	PkgPath          string         `json:",omitempty"` // This field is populated in the case of language-specific packages such as egg/wheel and gemspec
+	PkgIdentifier    *PkgIdentifier `json:",omitempty"`
 	InstalledVersion string         `json:",omitempty"`
 	FixedVersion     string         `json:",omitempty"`
 	Status           types.Status   `json:",omitempty"`
 	Layer            ftypes.Layer   `json:",omitempty"`
 	SeveritySource   types.SourceID `json:",omitempty"`
 	PrimaryURL       string         `json:",omitempty"`
 
-	// PkgRef is populated only when scanning SBOM and contains the reference ID used in the SBOM.
-	// It could be PURL, UUID, etc.
-	// e.g.
-	//    - pkg:npm/acme/component@1.0.0
-	//    - b2a46a4b-8367-4bae-9820-95557cfe03a8
-	PkgRef string `json:",omitempty"`
```

Comment:
We used to need it in order to store BOM-Ref. I'm afraid that removing this field affects scanning somewhere. But we may no longer need it since we dropped some capabilities. I'll take a look.

Reply:
I marked it as deprecated at https://github.com/aquasecurity/trivy/pull/5439/commits/7d29afd668566c2cd72eff54e0affc971a522c87 instead of removing it